pipelinewise/cli/pipelinewise.py

```diff
@@ -1621,6 +1658,171 @@ def encrypt_string(self):
+                tap_params = TapParams(
+                    tap_id=tap_id,
+                    type=tap_type,
+                    bin=self.tap_bin,
+                    python_bin=self.tap_python_bin,
+                    config=tap_config,
+                    properties=tap_properties,
+                    state=tap_state,
+                )
+
+                target_params = TargetParams(
+                    target_id=target_id,
+                    type=target_type,
+                    bin=self.target_bin,
+                    python_bin=self.target_python_bin,
+                    config=cons_target_config,
+                )
+
+                transform_params = TransformParams(
+                    bin=self.transform_field_bin,
+                    config=tap_transformation,
+                    python_bin=self.transform_field_python_bin,
+                    tap_id=tap_id,
+                    target_id=target_id,
+                )
+
+                self.run_tap_partialsync(
+                    tap=tap_params, target=target_params, transform=transform_params,
+                )
+
+        except pidfile.AlreadyRunningError as exc:
+            self.logger.error('Another instance of the tap is already running.')
+            raise SystemExit(1) from exc
+        # Delete temp file if there is any
+        except commands.RunCommandException as exc:
+            self.logger.exception(exc)
+            self.send_alert(message=f'Failed to sync tables in {tap_id} tap', exc=exc)
+            raise SystemExit(1) from exc
+        except PartialSyncNotSupportedTypeException as exc:
+            self.logger.error(exc)
+            raise SystemExit(1) from exc
+        except PreRunChecksException as exp:
+            raise SystemExit(1) from exp
+        except Exception as exc:
+            self.send_alert(message=f'Failed to sync tables in {tap_id} tap', exc=exc)
+            raise exc
+        finally:
+            if cons_target_config:
+                utils.silentremove(cons_target_config)
+
+    def _check_supporting_tap_and_target_for_partial_sync(self):
+        tap_type = self.tap['type']
+        tap_id = self.tap['id']
+        target_type = self.target['type']
+        target_id = self.target['id']
+
+        if ConnectorType(target_type) not in PARTIAL_SYNC_PAIRS.get(ConnectorType(tap_type), {}):
+            raise PartialSyncNotSupportedTypeException(
+                f'Error! {tap_id}({tap_type})-{target_id}({target_type}) pair is not supported for the partial sync!'
+            )
+
+    def _check_if_complete_tap_configuration(self, fastsync_bin, tap_type, target_type):
+        # Tap exists but configuration not completed
+        if not os.path.isfile(fastsync_bin):
+            self.logger.error(
+                'Table sync function is not implemented from %s datasources to %s type of targets',
+                tap_type,
+                target_type
+            )
+            raise SystemExit(1)
```

Comment:
I suggest raising a `NotImplementedError`, it's more proper for this check.

Reply:
These parts of the codes are just copied from the existing codes. raising different exception causes some other tests that are not related to this PR to be failed because it is used by other parts of the project. here the main big method is only broken into some small methods to be more clean.
Spending time to refactor the other parts that makes also some concerns which may cause to break some functionalities is out of scope.